### PR TITLE
Add waitForAlertToBePresent method

### PR DESCRIPTION
**Pull request by pedrohyvo**
Add `waitForAlertToBePresent` method.
Closes https://github.com/wlsf82/protractor-helper/issues/114

## Pull Request Checklist

- [x] Have you followed the steps in our [Contributing](https://github.com/wlsf82/protractor-helper/blob/master/docs/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wlsf82/protractor-helper/pulls) for the same update/change?
- [x] Have you successfully ran tests with your changes locally?
- [X] Have you written new tests for your changes, if necessary?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you made improvements to the documentation, if necessary? ([docs/](https://github.com/wlsf82/protractor-helper/tree/master/docs) folder and [README.md](https://github.com/wlsf82/protractor-helper/blob/master/README.md))
- [X] Have you linked the correct [GitHub issue](https://github.com/wlsf82/protractor-helper/issues), if there is any?

> Pull requests that do not pass the coding static analysis and the automated tests on [Github Actions](https://github.com/wlsf82/protractor-helper/actions), and the code quality verification on [Better Code Hub](https://bettercodehub.com/) will not be reviewed.

### CODE REVIEW

**Comment by wlsf82** [comment]
@pedrohyvo, you shouldn't check the checkboxes from the **Pull Request Checklist** until we really do what they say.

**Comment by wlsf82** [comment]
> @pedrohyvo, you shouldn't check the checkboxes from the **Pull Request Checklist** until we really do what they say.

I have unchecked those that have not been followed.

**Comment by wlsf82** [comment]
@pedrohyvo, I've converted your PR on a draft since it's not complete yet.

File: src/waiters.js [hunk 1]
```diff
@@ -3,6 +3,15 @@ const EC = protractor.ExpectedConditions;
 const messageBuilder = require("./constants_and_utils/messageBuilder");
 const utils = require("./constants_and_utils/utils");
 
+const waitForAlertToBePresent = function(
+  timeoutInMilliseconds = utils.timeout.timeoutInMilliseconds
+) {
+  return browser.wait(
```
Comment (wlsf82) [inline src/waiters.js:9]: @pedrohyvo, there's a missing parameter after the `timeoutInMilliseconds`, which is the message that should be shown if the alert does not present after the determined timeout.

Take a look at other methods in this file to understand how this is done.

File: test/e2e/sample.spec.js [hunk 2]
```diff
@@ -8,12 +8,18 @@ describe("Protractor helper", () => {
...
+  const alertButton = element(by.id("alertButton"));
 
   const nonExistingElement = element(by.id("foo"));
 
   beforeEach(() => browser.get(""));
 
   describe("waits", () => {
+    it("waitForAlertToBePresent", () => {
+      helper.click(alertButton);
+      helper.waitForAlertToBePresent(5000);
```
Comment (wlsf82) [inline test/e2e/sample.spec.js:20]:
```suggestion
      helper.waitForAlertToBePresent();
```

This first test should test that it works without passing the optional timeout argument, just to follow the order we've been using for other similar tests.

Comment (wlsf82) [reply test/e2e/sample.spec.js:20]: We also need to test if it works when passing a timeout different from the default one. See other tests for examples.

File: test/e2e/sampleApp/index.html [hunk 3]
```diff
@@ -41,6 +41,7 @@ <h1>Enter a URL</h1>
           />
           <label for="file"><strong>Choose a file</strong></label>
         </form>
+        <button id="alertButton" onclick="alert('Hello\nHow are you?')">Test alert</button>
```
Comment (wlsf82) [inline test/e2e/sampleApp/index.html:44]:
```suggestion
        <button id="alertButton" onclick="alert('Hey, I am a test alert.')">Click me!</button>
```

File: index.d.ts [hunk 4]
```diff
@@ -18,6 +18,7 @@ export function scrollToElement(elem: ElementFinder, timeoutInMs?: number): Prom
 export function setTimeout(timeoutInMs?: number): void
 
 // waiters
+export function waitForAlertToBePresent(timeoutInMs?: number): void
```
Comment (wlsf82) [inline index.d.ts:21]:
```suggestion
export function waitForAlertToBePresent(timeoutInMs?: number): Promise<T>
```

See http://www.protractortest.org/#/api?view=ProtractorExpectedConditions.prototype.alertIsPresent

File: test/e2e/sample.spec.js (hunk 2 above)
Comment (wlsf82) [reply test/e2e/sample.spec.js:20]: You will also need to accept the dialog that the test will trigger.

I suggest using something like:

```js
const alertDialog = browser.switchTo().alert();
alertDialog.accept();
```

**Review by wlsf82 — CHANGES_REQUESTED** [review]
Hi, @pedrohyvo. Thanks for another contribution!

I know this PR is still a draft, but I took the liberty to add a few suggestions and comments. I hope they help you! 💯

File: test/e2e/sample.spec.js (hunk 2 above)
Comment (pedrohyvo) [reply test/e2e/sample.spec.js:20]: Done

File: src/waiters.js (hunk 1 above)
Comment (pedrohyvo) [reply src/waiters.js:9]: Done

Comment (pedrohyvo) [reply src/waiters.js:9]: Done.

**Comment by pedrohyvo** [comment]
@wlsf82, can you review this PR again, please?

File: docs/EXAMPLES.md [hunk 5]
```diff
@@ -66,11 +66,33 @@ describe("foo", () => {
...
+| `timeoutInMilliseconds` (default: 5000) |
+
+```js
+const protractorHelper = require("protractor-helper");
+
+describe("foo", () => {
+  it("bar", () => {
+    browser.get("https://example.com");
+
+    const alertElement = element(by.className("el-alert"));
```
Comment (wlsf82) [inline docs/EXAMPLES.md:82]: Hey, @pedrohyvo, an alert can't be located as an HTML element.

Here we should have something like this.

```js
const buttonThatTriggersAnAlert = element(by.css('button.some-button'));

helper.click(buttonThatTriggersAnAlert);
```

Then what's already on line 84.

:wink:

File: README.md [hunk 6]
```diff
@@ -163,6 +164,11 @@ If called without passing an argument the timeout will be set to the default one
 This method returns a boolean depending if the current url is different from the base url. No argument needed.
 [Example](docs/EXAMPLES.md#iscurrenturldifferentfrombaseurl)
 
+### `waitForAlertToBePresent`
+
+This method waits for an alert to be present in the DOM.
```
Comment (wlsf82) [inline README.md:169]:
```suggestion
This method waits for an alert to be present on the Page.
```

File: docs/EXAMPLES.md [hunk 7]
```diff
@@ -14,8 +14,8 @@
 
 ## setTimeout
 
-|           1 optional argument           |
-| :-------------------------------------: |
+| 1 optional argument                     |
```
Comment (wlsf82) [inline docs/EXAMPLES.md:17]: @pedrohyvo, out-of-curiosity, why did you change that?

Same question for all other appearences of that.

File: docs/PT-BR/EXAMPLES.md [hunk 8]
```diff
@@ -66,11 +66,33 @@ describe("foo", () => {
...
+| `timeoutInMilliseconds` (padrão: 5000) |
+
+```js
+const protractorHelper = require("protractor-helper");
+
+describe("foo", () => {
+  it("bar", () => {
+    browser.get("https://example.com");
+
+    const alertElement = element(by.className("el-alert"));
```
Comment (wlsf82) [inline docs/PT-BR/EXAMPLES.md:82]: This needs to be adjusted as in the English version.

File: docs/PT-BR/README.md [hunk 9]
```diff
@@ -160,6 +160,11 @@ Este método retorna um valor booleano dependendo se a URL atual é diferente da
 
 [Exemplo](./EXAMPLES.md#iscurrenturldifferentfrombaseurl)
 
+### `waitForAlertToBePresent`
+
+Este método espera até que um alerta esteja presente no DOM.
```
Comment (wlsf82) [inline docs/PT-BR/README.md:165]:
```suggestion
Este método espera até que um alerta esteja presente na página.
```

File: src/constants_and_utils/messageBuilder.js [hunk 10]
```diff
@@ -1,5 +1,9 @@
 const constants = require("./constants");
 
+function getDefaultAlertIsNotVisible() {
```
Comment (wlsf82) [inline src/constants_and_utils/messageBuilder.js:3]: Shouldn't it be `getDefaultAlertIsNotPresent` instead?

File: src/constants_and_utils/messageBuilder.js [hunk 11]
```diff
@@ -1,5 +1,9 @@
...
+  return `alert ${constants.IS_NOT_VISIBLE_MESSAGE}.`;
```
Comment (wlsf82) [inline src/constants_and_utils/messageBuilder.js:4]:
```suggestion
  return `alert ${constants.IS_NOT_PRESENT_MESSAGE}.`;
```

File: src/constants_and_utils/messageBuilder.js [hunk 12]
```diff
@@ -52,7 +56,7 @@ function getDefaultIsStillVisibleMessage(htmlElement) {
   }.`;
 }
 
-function getDeafultTextTextIsStillPresentOnElementMessage(htmlElement, text) {
+function getDefaultTextTextIsStillPresentOnElementMessage(htmlElement, text) {
```
Comment (wlsf82) [inline src/constants_and_utils/messageBuilder.js:59]: It seems that there's another typo here 😜 

```suggestion
function getDefaultTextIsStillPresentOnElementMessage(htmlElement, text) {
```

File: src/constants_and_utils/messageBuilder.js [hunk 13]
```diff
@@ -81,7 +86,7 @@ module.exports = {
   getDefaultIsStillPresentMessage,
   getDefaultIsNotTappableMessage,
   getDefaultIsStillVisibleMessage,
-  getDeafultTextTextIsStillPresentOnElementMessage,
+  getDefaultTextTextIsStillPresentOnElementMessage,
```
Comment (wlsf82) [inline src/constants_and_utils/messageBuilder.js:89]:
```suggestion
  getDefaultTextIsStillPresentOnElementMessage,
```

File: src/waiters.js [hunk 14]
```diff
@@ -3,6 +3,16 @@ const EC = protractor.ExpectedConditions;
...
+    EC.alertIsPresent(),
+    timeoutInMilliseconds,
+    messageBuilder.getDefaultAlertIsNotVisible()
```
Comment (wlsf82) [inline src/waiters.js:12]:
```suggestion
    messageBuilder.getDefaultAlertIsNotPresent()
```

File: src/waiters.js [hunk 15]
```diff
@@ -67,7 +77,7 @@ const waitForTextNotToBePresentInElement = function(
   return browser.wait(
     EC.not(EC.textToBePresentInElement(htmlElement, text)),
     timeoutInMilliseconds,
-    messageBuilder.getDeafultTextTextIsStillPresentOnElementMessage(htmlElement, text)
+    messageBuilder.getDefaultTextTextIsStillPresentOnElementMessage(htmlElement, text)
```
Comment (wlsf82) [inline src/waiters.js:80]: Same as https://github.com/wlsf82/protractor-helper/pull/140/files#r563871744

File: test/e2e/sample.spec.js [hunk 16]
```diff
@@ -8,12 +8,27 @@ describe("Protractor helper", () => {
   const inputFile = element(by.className("inputfile"));
   const shortenButton = element(by.id("shorten"));
   const expandButton = element(by.id("expand"));
+  const alertButton = element(by.id("alertButton"));
```
Comment (wlsf82) [inline test/e2e/sample.spec.js:11]: What about

```suggestion
  const buttonThatTriggersAnAlert = element(by.id("alertButton"));
```

File: test/e2e/sample.spec.js [hunk 17]
```diff
@@ -8,12 +8,27 @@ describe("Protractor helper", () => {
...
 
   const nonExistingElement = element(by.id("foo"));
 
   beforeEach(() => browser.get(""));
 
   describe("waits", () => {
+    it("waitForAlertToBePresent", () => {
+      helper.click(alertButton);
```
Comment (wlsf82) [inline test/e2e/sample.spec.js:19]:
```suggestion
      helper.click(buttonThatTriggersAnAlert);
```

File: test/e2e/sample.spec.js [hunk 18]
```diff
@@ -8,12 +8,27 @@ describe("Protractor helper", () => {
...
+      helper.waitForAlertToBePresent();
+      const alertDialog = browser.switchTo().alert();
+      alertDialog.accept();
+    });
+
+    it("waitForAlertToBePresent with different timeout", () => {
+      helper.click(alertButton);
```
Comment (wlsf82) [inline test/e2e/sample.spec.js:26]: Same as https://github.com/wlsf82/protractor-helper/pull/140/files#r563872988

File: test/spec/messageBuilder.spec.js [hunk 19]
```diff
@@ -133,7 +140,7 @@ describe("messageBuilder", () => {
     expect(actualResult).toEqual(expectedResult);
   });
 
-  it("getDeafultTextTextIsStillPresentOnElementMessage()", () => {
+  it("getDefaultTextTextIsStillPresentOnElementMessage()", () => {
```
Comment (wlsf82) [inline test/spec/messageBuilder.spec.js:143]:
```suggestion
  it("getDefaultTextIsStillPresentOnElementMessage()", () => {
```

File: test/spec/messageBuilder.spec.js [hunk 20]
```diff
@@ -143,7 +150,7 @@ describe("messageBuilder", () => {
     const someText = "some text";
 
     const actualResult = messageBuilder
-      .getDeafultTextTextIsStillPresentOnElementMessage(sampleElement, someText);
+      .getDefaultTextTextIsStillPresentOnElementMessage(sampleElement, someText);
```
Comment (wlsf82) [inline test/spec/messageBuilder.spec.js:153]:
```suggestion
      .getDefaultTextIsStillPresentOnElementMessage(sampleElement, someText);
```

**Review by wlsf82 — CHANGES_REQUESTED** [review]
Great work, @pedrohyvo!

I have left some comments and suggestions that need to be addressed.

As soon as you're done with them, please ask me for another round of review.

👍🏻

File: docs/EXAMPLES.md (hunk 5 above)
Comment (pedrohyvo) [reply docs/EXAMPLES.md:82]: Done.

File: README.md (hunk 6 above)
Comment (pedrohyvo) [reply README.md:169]: done

File: docs/EXAMPLES.md (hunk 7 above)
Comment (pedrohyvo) [reply docs/EXAMPLES.md:17]: @wlsf82 I've formatted the code using a VS code plugin `Markdown Table Prettifier`.

Comment (pedrohyvo) [reply docs/EXAMPLES.md:17]: done

File: docs/PT-BR/EXAMPLES.md (hunk 8 above)
Comment (pedrohyvo) [reply docs/PT-BR/EXAMPLES.md:82]: done

File: docs/PT-BR/README.md (hunk 9 above)
Comment (pedrohyvo) [reply docs/PT-BR/README.md:165]: done

File: src/constants_and_utils/messageBuilder.js (hunk 11 above)
Comment (pedrohyvo) [reply src/constants_and_utils/messageBuilder.js:4]: done

File: src/constants_and_utils/messageBuilder.js (hunk 12 above)
Comment (pedrohyvo) [reply src/constants_and_utils/messageBuilder.js:59]: done

File: src/constants_and_utils/messageBuilder.js (hunk 13 above)
Comment (pedrohyvo) [reply src/constants_and_utils/messageBuilder.js:89]: done

File: src/waiters.js (hunk 14 above)
Comment (pedrohyvo) [reply src/waiters.js:12]: done

File: src/waiters.js (hunk 15 above)
Comment (pedrohyvo) [reply src/waiters.js:80]: done

File: test/e2e/sample.spec.js (hunk 16 above)
Comment (pedrohyvo) [reply test/e2e/sample.spec.js:11]: done

**Comment by pedrohyvo** [comment]
@wlsf82, can you review again this PR, please?

File: docs/EXAMPLES.md [hunk 21]
```diff
@@ -14,8 +14,8 @@
 
 ## setTimeout
 
-| 1 optional argument                     |
-|-----------------------------------------|
+|           1 optional argument           |
```
Comment (wlsf82) [inline docs/EXAMPLES.md]: You didn't necessarily have to change it back. I'm fine with both ways. It's on you!
I just asked you why you changed it? By the way, I'm still curious about that. 😜

File: docs/EXAMPLES.md [hunk 22]
```diff
@@ -68,18 +68,18 @@ describe("foo", () => {
...
+| :-------------------------------------: |
 | `timeoutInMilliseconds` (default: 5000) |
 
 ```js
 const protractorHelper = require("protractor-helper");
 
 describe("foo", () => {
   it("bar", () => {
-    browser.get("https://example.com");
```
Comment (wlsf82) [inline docs/EXAMPLES.md]: Visiting the page should still happen.

File: docs/PT-BR/EXAMPLES.md [hunk 23]
```diff
@@ -79,7 +79,7 @@ describe("foo", () => {
   it("bar", () => {
     browser.get("https://example.com");
 
-    const alertElement = element(by.className("el-alert"));
+    const elementoAlerta = element(by.className("elemento-alerta"));
```
Comment (wlsf82) [inline docs/PT-BR/EXAMPLES.md:82]: This part should reflect the same thing as in the English version. And we don't translate the `code.`

**Review by wlsf82 — CHANGES_REQUESTED** [review]
@pedrohyvo, thanks again!

I left three more comments that need to be addressed before merging your PR.

Can you fix them, please?

File: docs/EXAMPLES.md (hunk 21 above)
Comment (pedrohyvo) [reply docs/EXAMPLES.md]: I just formatted using a markdown plugin 🤷‍♂️  that's why I prefer to revert back 😂

File: docs/EXAMPLES.md (hunk 22 above)
Comment (pedrohyvo) [reply docs/EXAMPLES.md]: done

File: docs/PT-BR/EXAMPLES.md (hunk 23 above)
Comment (pedrohyvo) [reply docs/PT-BR/EXAMPLES.md:82]: done

**Comment by pedrohyvo** [comment]
> @pedrohyvo, thanks again!
> 
> I left three more comments that need to be addressed before merging your PR.
> 
> Can you fix them, please?

@wlsf82 sure, can you review again, please? Hope it's good to go now.

**Review by wlsf82 — CHANGES_REQUESTED** [review]
@pedrohyvo, you forgot to add the new method on the summary of the available functions on `docs/PT-BR/README.md`

**Comment by pedrohyvo** [comment]
@wlsf82 added right now 👍

File: docs/EXAMPLES.md [hunk 24]
```diff
@@ -66,11 +66,35 @@ describe("foo", () => {
...
+```js
+const protractorHelper = require("protractor-helper");
+
+describe("foo", () => {
+  it("bar", () => {
+    browser.get("https://example.com");
+
+    const buttonThatTriggersAnAlert = element(by.css('button.some-button'));
+
+    helper.click(buttonThatTriggersAnAlert);
```
Comment (wlsf82) [inline docs/EXAMPLES.md:84]:
```suggestion
    protractorHelper.click(buttonThatTriggersAnAlert);
```

Oh gosh, I just noticed that. Sorry for not finding it before. 🤦🏻‍♂️ 

On line 76, we're importing protractor-helper as `protractorHelper`, and so, this is the variable that should be used for the `click` too.

File: docs/PT-BR/EXAMPLES.md [hunk 25]
```diff
@@ -66,11 +66,35 @@ describe("foo", () => {
...
+```js
+const protractorHelper = require("protractor-helper");
+
+describe("foo", () => {
+  it("bar", () => {
+    browser.get("https://example.com");
+
+    const buttonThatTriggersAnAlert = element(by.css('button.some-button'));
+
+    helper.click(buttonThatTriggersAnAlert);
```
Comment (wlsf82) [inline docs/PT-BR/EXAMPLES.md:84]: Same as https://github.com/wlsf82/protractor-helper/pull/140/files#r564858848

```suggestion
    protractorHelper.click(buttonThatTriggersAnAlert);
```

**Review by wlsf82 — CHANGES_REQUESTED** [review]
@pedrohyvo, I found another change that I didn't realize before. I think this will be the last one 😉 
Thanks for your patience.

**Comment by pedrohyvo** [comment]
@wlsf82, can you review this PR again, please

**Review by wlsf82 — APPROVED** [review]
@pedrohyvo, you've got a :cake:

Thanks for another contribution!
